en/dev_log/log_encryption.md

```diff
@@ -0,0 +1,179 @@
+# Log Encryption
+
+The [System Logger](../modules/modules_system.md#logger) can be used to create encrypted logs, which may then be decrypted manually before analysis.
+
+The encryption algorithm used is XChaCha20, and the default wrapping algorithm used is RSA2048-OAEP.
+
+::: warning
+Log encryption is not enabled by default in PX4 firmware builds.
+To use it you will need to build firmware with this feature enabled and then upload it to the flight controller (see instructions below).
+:::
+
+::: info
+The encryption algorithm used can be changed by modifying [SDLOG_ALGORITHM](../advanced_config/parameter_reference.md#SDLOG_ALGORITHM).
+However at time of writing, only XChaCha20 can actually be used (AES can be selected, but there is no implementation).
+:::
+
+## How ULog Encryption Works
+
+When the ULog file is first opened:
+
+- If [SDLOG_ALGORITHM](../advanced_config/parameter_reference.md#SDLOG_ALGORITHM) is set to default, a symmetric XChaCha20 key is generated and wrapped (encrypted) with RSA2048 public key encryption
+- This wrapped key is stored on the SD Card with the ".ulgk" suffix.
+
+While the ULog file is written to disk:
+
+- The generated (plaintext) key is used to encrypt the data blocks right before writing to disk.
+
+After the flight, there are two files:
+
+- `.ulogc` (ulog cipher): the actual log file data, encrypted
+- `.ulogk` (ulog wrapped key): the symmetric key, wrapped with rsa
+
+However, these files will both have the normal `.ulg` suffix if downloading from QGroundControl.
+
+## Creating a Flight Controller Build that contains Log Encryption
+
+Crypto uses large amounts of flash memory, and is therefore not included in PX4 builds by default (the exception is the make target `px4-fmu-v5_cryptotest`).
+Below we show how to add log encryption functionality to a flight controller target, using the `px4-fmu-v5` board as an example.
+
+Reference the `cryptotest.px4board` file in `boards/px4/fmu-v5` for the arguments you will need to add to the `.px4board` file associated with the flight controller you are targeting.
+You could also copy and paste `cryptotest.px4board` to the other flight controller's `boards/` location if you want this to be a separate build entirely (in that case, your `make` command would include `_cryptotest` at the end to add log encryption).
```

Comment:
I think you are saying in the second line "You could" here, that _generically_ if there is a config file with name `_foo.px4board` in `boards/px4/fmu-v6` (say) you could build  `make px4-fmu-v6_foo` and that would generate firmware `px4-fmu-v5_foo` - right?
So in this case you're saying that you can build firmware with this name by copying the config file for cryptotest.

In the first line are you saying you should just copy the content of tge `cryptotest.px4board` file into the default board config?

Why I am confused is why you would do the copying. I THINK that you need to do this before you do the "Adding log encryption to NuttX config" so that the options appear in the kconfig UI. Is that correct.

So the steps are "add info about the config variables" to the config folder, then run kconfig and set their values. Right?

Reply:
> I think you are saying in the second line "You could" here, that generically if there is a config file with name _foo.px4board in boards/px4/fmu-v6 (say) you could build make px4-fmu-v6_foo and that would generate firmware px4-fmu-v5_foo - right?

I think my wording here was a bit confusing. I was essentially trying to illustrate the point that you don't HAVE to copy and paste the cryptotest.px4board file to a different board type, and instead, you could just copy the parameters in that file into a different one. 

> Why I am confused is why you would do the copying.

If you have multiple builds for a particular board and want your log encryption build to be separate from other builds, it would make sense to copy the cryptotest.px4board file to the other board so you can make it by doing make {board}_cryptotest. As an example, if you get rid of a feature due to flash memory constraints, you may want multiple variations of the firmware build so that you can keep that feature in some.

Is it necessary for us to mention this in the documentation? Probably not. 

> So the steps are "add info about the config variables" to the config folder, then run kconfig and set their values. Right?

Yes, but I don't believe the order at which you do so matters.